core/services/relay/evm/mercury_provider.go

```diff
@@ -132,7 +129,7 @@ func (p *mercuryProvider) MercuryServerFetcher() mercurytypes.ServerFetcher {
 }
 
 func (p *mercuryProvider) ChainReader() commontypes.ContractReader {
-	return p.chainReader
+	return nil
```

Comment:
Should this panic? Or better, remove the `ChainReader` function from the provider interface? Returning nil might cause confusion or errors downstream where a reader is expected and no nil checks are done.

Reply:
This is coming from the PluginProvider, all other product providers just return nil